Yoast/wordpress-seo
# Deprecate yoast_plugin_compatibility_notification and related code

## Summary


This PR can be summarized in the following changelog entry:

* Removes the notification that would be thrown if add-ons with a lower version number than the plugin were installed. The `yoast_plugin_compatibility_notification` method has been deprecated.

## Relevant technical choices:

* 

## Test instructions

This PR can be tested by following these steps:

* Install add-on versions below 12.2.
* Install the Yoast test helper.
* Go to trunk (or if RC-testing: use a pre-12.3 zip) of wordpress-seo. Build everything, and go to the Yoast Notification center. See that there are notifications under `Problems` about old add-on versions.
* Switch to this branch.
* If acceptance testing this: change `if ( version_compare( $version, '12.3-RC0', '<' ) ) {` in `inc/class-upgrade.php` to `if ( version_compare( $version, '12.1-RC0', '<' ) ) {`
    * You don't need to do this step if you're RC-testing the 12.3-RCx zip.
* In the test helper, set the database to an old version, e.g. 10.0, and click `save`. This will trigger the upgrade routine.
* Go back to the dashboard and see that the notifications are gone.

## UI changes

## Documentation

## Quality assurance


Fixes #13485
